extensions/librdkafka/rdkafka_utils.h

```diff
@@ -47,7 +47,7 @@ struct rd_kafka_conf_deleter {
 
 struct rd_kafka_producer_deleter {
   void operator()(rd_kafka_t* ptr) const noexcept {
-    rd_kafka_resp_err_t flush_ret = rd_kafka_flush(ptr, 10000 /* ms */);  // Matching the wait time of KafkaConnection.cpp
+    // rd_kafka_resp_err_t flush_ret = rd_kafka_flush(ptr, 10000 /* ms */);  // Matching the wait time of KafkaConnection.cpp
```

Comment:
Why did you comment this out?
This is part of the proper termination sequence:
https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md#termination

Reply:
My mistake, the problem was the unused parameter and I commented it out without realizing that we need the flush call. Thanks for the info.

Reply:
Updated in [75e1aa](https://github.com/apache/nifi-minifi-cpp/pull/1020/commits/75e1aa33766b2b43b135ebfd2aac312e796d463a)